File: repos_estudiantes_TARQUI_tercera_entrega/SIÑANI_QUELALI_IVAN/tarea3/per1/archivo_empleado.py
import pickle
from empleado import Empleado  

class ArchivoEmpleado:
    def __init__(self, nombre_archivo):
        self.nombre_archivo = nombre_archivo

    def crear_archivo(self):
        """Crea el archivo si no existe"""
        try:
            with open(self.nombre_archivo, "wb") as archivo:
                pickle.dump([], archivo) 
        except Exception as e:
            print("Error al crear archivo:", e)

    def guardar_empleado(self, empleado):
        """Guarda un empleado en el archivo"""
        empleados = self.leer_todos()
        empleados.append(empleado)
        with open(self.nombre_archivo, "wb") as archivo:
            pickle.dump(empleados, archivo)

    def leer_todos(self):
        """Lee todos los empleados guardados en el archivo"""
        try:
            with open(self.nombre_archivo, "rb") as archivo:
                return pickle.load(archivo)
        except (FileNotFoundError, EOFError):
            return []

    def busca_empleado(self, nombre):
        """Busca un empleado por nombre"""
        empleados = self.leer_todos()
        for emp in empleados:
            if emp.nombre == nombre:
                return emp
        return None

    def mayor_salario(self, salario):
        """Encuentra el primer empleado con salario mayor al indicado"""
        empleados = self.leer_todos()
        for emp in empleados:
            if emp.salario > salario:
                return emp
        return None

File: repos_estudiantes_TARQUI_tercera_entrega/MACHACA_CALLATA_CARLOS_GHILMER_CARLOS_GHILMER/Persistencia/Ejer1/codigo/ArchivoEmpleado.java

package Persistencia.Ejer1.codigo;

import java.io.File;
import java.io.IOException;
import java.util.ArrayList;

public class ArchivoEmpleado {
    private String nomA;
    ArrayList<Empleado> empleados;

    public ArchivoEmpleado(String nomA) {
        this.nomA = nomA;
        this.empleados = new ArrayList<>();
    }
    public void crearArchivo() {
        try {
            File archivo = new File(nomA);
            if (archivo.createNewFile()) {
                System.out.println("Archivo creado: " + archivo.getName());
            } else {
                System.out.println("El archivo ya existe.");
            }
        } catch (IOException e) {
            System.out.println("Error al crear archivo.");
            e.printStackTrace();
        }
    }
    public void guardarEmpleado(Empleado e){
        empleados.add(e);
    }
    public Empleado buscaEmpleado(String n) {
        for(Empleado e : empleados) {
            if(e.getNombre().equals(n)) return e;
        }
        return null;
    }
     public Empleado mayorSalario(float sueldo) {
        Empleado mayor = null;
        for (Empleado e : empleados) {
            if (e.getSalario() > sueldo) {
                if (mayor == null || e.getSalario() > mayor.getSalario()) {
                    mayor = e;
                }
            }
        }
        return mayor;
    }

}